sample/src/main/java/com/uber/autodispose/recipes/AutoDisposeActivity.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2017. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.autodispose.recipes;
+
+import android.app.Activity;
+import android.os.Bundle;
+import com.uber.autodispose.LifecycleEndedException;
+import com.uber.autodispose.LifecycleScopeProvider;
+import io.reactivex.Observable;
+import io.reactivex.functions.Function;
+import io.reactivex.subjects.BehaviorSubject;
+import javax.annotation.Nullable;
+
+/**
+ * An {@link Activity} example implementation for making one implement {@link
+ * LifecycleScopeProvider}. One would normally use this as a base activity class to extend others
+ * from.
+ */
+public abstract class AutoDisposeActivity extends Activity
+    implements LifecycleScopeProvider<AutoDisposeActivity.ActivityEvent> {
+
+  public enum ActivityEvent {
+    CREATE, START, RESUME, PAUSE, STOP, DESTROY
+  }
+
+  private static Function<ActivityEvent, ActivityEvent> CORRESPONDING_EVENTS =
+      new Function<ActivityEvent, ActivityEvent>() {
+        @Override public ActivityEvent apply(ActivityEvent activityEvent) throws Exception {
+          switch (activityEvent) {
```

Comment:
I'd split these events into two separate sequences, so it'll be easier to understand, why startup methods have symmetric shutdown ones as a counterpart, while shutdown methods are having sequential shutdown ones.